File: Habitica/src/main/java/com/habitrpg/android/habitica/data/local/implementation/RealmFAQLocalRepository.java
package com.habitrpg.android.habitica.data.local.implementation;

import com.habitrpg.android.habitica.data.local.FAQLocalRepository;
import com.habitrpg.android.habitica.models.FAQArticle;

import io.realm.Realm;
import io.realm.RealmResults;
import rx.Observable;

public class RealmFAQLocalRepository extends RealmBaseLocalRepository implements FAQLocalRepository {

    public RealmFAQLocalRepository(Realm realm) {
        super(realm);
    }

    @Override
    public Observable<RealmResults<FAQArticle>> getArticles() {
        return realm.where(FAQArticle.class)
                .findAll()
                .asObservable()
                .filter(RealmResults::isLoaded);
    }
}

File: Habitica/src/main/java/com/habitrpg/android/habitica/data/local/implementation/RealmFAQLocalRepository.kt
package com.habitrpg.android.habitica.data.local.implementation

import com.habitrpg.android.habitica.data.local.FAQLocalRepository
import com.habitrpg.android.habitica.models.FAQArticle
import io.realm.Realm
import kotlinx.coroutines.flow.Flow

class RealmFAQLocalRepository(
    realm: Realm,
) : RealmContentLocalRepository(realm),
    FAQLocalRepository {
    override fun getArticle(position: Int): Flow<FAQArticle> = safeFindOne {
        it.where(FAQArticle::class.java).equalTo("position", position)
    }

    override val articles: Flow<List<FAQArticle>>
        get() = safeFindAll { it.where(FAQArticle::class.java) }
}
